tests/utils/exec.ts

```diff
@@ -180,219 +198,134 @@ async function pkStdio(
+  } else {
+    return await pkSpawnWithShell(args, opts, logger);
+  }
 }
 
 /**
- * Mimics the behaviour of `pkStdio` while running the command as a separate process.
- * Note that this is incompatible with jest mocking.
- * @param args - args to be passed to the command.
- * @param env - environment variables to be passed to the command.
- * @param cwd - the working directory the command will be executed in.
+ * Runs pk command through subprocess
+ * This is the default
  */
-async function pkStdioTarget(
+async function pkExecWithoutShell(
   args: Array<string> = [],
-  env: Record<string, string | undefined> = {},
-  cwd?: string,
+  opts: ExecOpts = { env: {} },
 ): Promise<{
   exitCode: number;
   stdout: string;
   stderr: string;
 }> {
-  cwd = path.resolve(
-    cwd ??
-      (await fs.promises.mkdtemp(
-        path.join(globalThis.tmpDir, 'polykey-test-'),
-      )),
-  );
+  const cwd =
+    opts.cwd ??
+    (await fs.promises.mkdtemp(path.join(globalThis.tmpDir, 'polykey-test-')));
+  const env = {
+    ...process.env,
+    ...opts.env,
+  };
   // Recall that we attempt to connect to all specified seed nodes on agent start.
   // Therefore, for testing purposes only, we default the seed nodes as empty
   // (if not defined in the env) to ensure no attempted connections. A regular
   // PolykeyAgent is expected to initially connect to the mainnet seed nodes
   env['PK_SEED_NODES'] = env['PK_SEED_NODES'] ?? '';
-
-  // If using the command override we need to spawn a process
-  env = {
-    ...process.env,
-    ...env,
-    DOCKER_OPTIONS: generateDockerArgs(cwd).join(' '),
-  };
-  const command = globalThis.testCmd!;
-  const escapedArgs = args.map((x) => x.replace(/(["\s'$`\\])/g, '\\$1'));
-  const subprocess = child_process.spawn(command, escapedArgs, {
-    env,
-    cwd,
-    stdio: ['pipe', 'pipe', 'pipe'],
-    windowsHide: true,
-    shell: true,
-  });
-  const exitCodeProm = promise<number | null>();
-  subprocess.on('exit', (code) => {
-    exitCodeProm.resolveP(code);
-  });
-  subprocess.on('error', (e) => {
-    exitCodeProm.rejectP(e);
-  });
-  let stdout = '',
-    stderr = '';
-  subprocess.stdout.on('data', (data) => {
-    stdout += data.toString();
-  });
-  subprocess.stderr.on('data', (data) => {
-    stderr += data.toString();
+  return new Promise((resolve, reject) => {
+    let stdout = '',
+      stderr = '';
+    const subprocess = childProcess.spawn(
+      'ts-node',
+      ['--project', tsConfigPath, polykeyPath, ...args],
+      {
+        env,
+        cwd,
+        windowsHide: true,
+        shell: opts.shell ? opts.shell : false,
+      },
+    );
+    subprocess.stdout.on('data', (data) => {
+      stdout += data.toString();
+    });
+    subprocess.stderr.on('data', (data) => {
+      stderr += data.toString();
+    });
+    subprocess.on('exit', (code) => {
+      resolve({ exitCode: code ?? -255, stdout, stderr });
+    });
+    subprocess.on('error', (e) => {
+      reject(e);
+    });
```

Comment:
Is there any overlap betwee non-0 exit codes and the error event?

Reply:
Here it is, this is the example of what I'm talking about. So `testUtils.exec` and `testUtils.pkExec` should be similar in this regard.

Reply:
Yep, `testUtils.exec` now follows this same format.